$NetBSD: patch-configure,v 1.2 2023/12/17 13:59:38 nros Exp $

Always link shared. Honour LDFLAGS.
Install manpages on solaris just like on 
the other systems.

--- configure.orig	2014-11-09 22:15:29.000000000 +0000
+++ configure
@@ -4161,9 +4161,9 @@ case $target in
 *solaris*)
   $as_echo "#define __EXTENSIONS__ 1" >>confdefs.h
 
-  SHARED_EXT=".so.${MAJOR_VER}"
-  SHARED_ALT=".so"
-  LINK_SHARED="$LD"' -G -M $$(srcdir)/libtecla.map -o $$@ -h $$(@F) -z defs -i $$(LIB_OBJECTS) $$(LIBS) -lc'
+  SHARED_EXT=".so.${MAJOR_VER}.${MINOR_VER}.${MICRO_VER}"
+  SHARED_ALT=".so .so.${MAJOR_VER}"
+  LINK_SHARED="$LD"' $$(LDFLAGS) -G -M $$(srcdir)/libtecla.map -o $$@ -h $$(@F) -z defs -i $$(LIB_OBJECTS) $$(LIBS) -lc'
   SHARED_CFLAGS="-Kpic"
   case $CC in
   */cc|cc) SHARED_CFLAGS="$SHARED_CFLAGS -xstrconst" ;;
@@ -4171,19 +4171,8 @@ case $target in
   case $target_cpu in
   sparc) SHARED_CFLAGS="$SHARED_CFLAGS -xregs=no%appl"
   esac
-  case $target_os in
-  solaris2.[89]*|solaris2.1[0-9]*)
-    LIBR_MANEXT=3lib
-    FUNC_MANEXT=3tecla
-    LIBR_MANDIR=man$LIBR_MANEXT
-    FUNC_MANDIR=man$FUNC_MANEXT
-  esac
-  MISC_MANDIR="man5"
-  MISC_MANEXT="5"
-  FILE_MANDIR="man4"
-  FILE_MANEXT="4"
   ;;
-*linux*)
+*linux*|*netbsd*)
   SHARED_EXT=".so.${MAJOR_VER}.${MINOR_VER}.${MICRO_VER}"
   SHARED_ALT=".so .so.${MAJOR_VER}"
 
@@ -4211,7 +4200,7 @@ $as_echo "$tecla_cv_gnu_ld_script" >&6;
     VERSION_OPT=''
   fi
 
-  LINK_SHARED="$LD"' -o $$@ -soname libtecla$$(SUFFIX).so.'${MAJOR_VER}' -shared '$VERSION_OPT' $$(LIB_OBJECTS) $$(LIBS) -lc'
+  LINK_SHARED="$LD"' $$(LDFLAGS) -o $$@ -soname libtecla$$(SUFFIX).so.'${MAJOR_VER}' -shared '$VERSION_OPT' $$(LIB_OBJECTS) $$(LIBS) -lc'
   SHARED_CFLAGS="-fpic"
 
 
@@ -4230,7 +4219,7 @@ $as_echo "$tecla_cv_gnu_ld_script" >&6;
 *darwin*)
   SHARED_EXT=".${MAJOR_VER}.${MINOR_VER}.${MICRO_VER}.dylib"
   SHARED_ALT=".dylib .${MAJOR_VER}.dylib"
-  LINK_SHARED='$(CC) -o $$@ -dynamiclib -flat_namespace -undefined suppress -compatibility_version '${MAJOR_VER}.${MINOR_VER}' -current_version '${MAJOR_VER}.${MINOR_VER}.${MICRO_VER}' -install_name '${libdir}'/$$@ $$(LIB_OBJECTS)'
+  LINK_SHARED='$(CC) $$(LDFLAGS) -o $$@ -dynamiclib -flat_namespace -undefined suppress -compatibility_version '${MAJOR_VER}.${MINOR_VER}' -current_version '${MAJOR_VER}.${MINOR_VER}.${MICRO_VER}' -install_name '${libdir}'/$$@ $$(LIB_OBJECTS)'
   SHARED_CFLAGS=""
   ;;
 *dec-osf*)
@@ -4241,7 +4230,7 @@ $as_echo "$tecla_cv_gnu_ld_script" >&6;
   SHARED_EXT=".so.${MAJOR_VER}"
   SHARED_ALT=".so"
   VERSION_OPT='--version-script=$$(srcdir)/libtecla.map'
-  LINK_SHARED='ld -o $$@ -soname libtecla$$(SUFFIX).so.'${MAJOR_VER}' -shared '$VERSION_OPT' $$(LIB_OBJECTS) $$(LIBS) -lc'
+  LINK_SHARED='ld $$(LDFLAGS) -o $$@ -soname libtecla$$(SUFFIX).so.'${MAJOR_VER}' -shared '$VERSION_OPT' $$(LIB_OBJECTS) $$(LIBS) -lc'
   SHARED_CFLAGS="-fpic"
   ;;
 mips-sgi-irix*)
